### ISSUE TITLE
OneCycleLRWithWarmup does not starts with init_lr

### ISSUE THREAD

**Issue by lokeshkvn** [catalyst-team/catalyst#851]
## 🐛 Bug Report

OneCycleLRWithWarmup starts ahead of initial LR (does not start with init_lr)

If the number of steps is 6 and warmup_fraction = 0.5 and init_lr = 1e-3 and lr_range=(2e-3,1e-3),
The lr starts with 1.36-3 excutes two steps and starts cooling down from 2e-3

### How To Reproduce
1. Go to https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/segmentation-tutorial.ipynb#scrollTo=OG6_pgmkxk_b
2.
```
batch_size = 8
loaders = get_loaders(
    images=ALL_IMAGES[:200],
    masks=ALL_MASKS[:200],
    random_state=SEED,
    train_transforms_fn=train_transforms,
    valid_transforms_fn=valid_transforms,
    batch_size=batch_size
) 
```
3. set epochs to 10 
4. replace scheduler with 
```from catalyst.contrib.nn.schedulers.onecycle import OneCycleLRWithWarmup
scheduler = OneCycleLRWithWarmup(optimizer, num_steps = 6,lr_range=(2e-3,1e-3), init_lr = 1e-3, warmup_fraction = 0.5)
```
5. Run 

 

#### Screenshots

<img width="849" alt="Screenshot 2020-06-23 at 1 44 42 AM" src="https://user-images.githubusercontent.com/16884938/85318756-20e2ab80-b4f3-11ea-8589-40039d7a056f.png">

### Expected behavior

OneCycleLRWithWarmup should start with lr = init_lr
(Pardon me if I am wrong, please advise me if this is the correct behavior of OneCycleLRWithWarmup as I am new to DL)

### Environment

Google Colab

**Comment by Scitator** [comment]
hi @lokeshkvn , thanks for the issue!
Due to a bunch of current tasks (we are working on huge codestyle and docs update),
could you please help us and provide full-featured example for reproduce the issue?
here is the template:
```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from catalyst.dl import SupervisedRunner

# data
num_samples, num_features = int(1e4), int(1e1)
X, y = torch.rand(num_samples, num_features), torch.rand(num_samples)
dataset = TensorDataset(X, y)
loader = DataLoader(dataset, batch_size=32, num_workers=1)
loaders = {"train": loader, "valid": loader}

# model, criterion, optimizer, scheduler
model = torch.nn.Linear(num_features, 1)
criterion = torch.nn.MSELoss()
optimizer = torch.optim.Adam(model.parameters())
scheduler = torch.optim.lr_scheduler.MultiStepLR(optimizer, [3, 6])

# model training
runner = SupervisedRunner()
runner.train(
    model=model,
    criterion=criterion,
    optimizer=optimizer,
    scheduler=scheduler,
    loaders=loaders,
    logdir="./logdir",
    num_epochs=8,
    verbose=True,
)
```

**Comment by lokeshkvn** [comment]
Sure @Scitator I will update you with the full-feature example.

**Comment by lokeshkvn** [comment]
Hi @Scitator  

Here is the code example to reproduce:

```python
import torch
from torch.utils.data import DataLoader, TensorDataset
from catalyst.dl import SupervisedRunner
from catalyst.contrib.nn.schedulers.onecycle import OneCycleLRWithWarmup

# data
num_samples, num_features = int(1e4), int(1e1)
X, y = torch.rand(num_samples, num_features), torch.rand(num_samples)
dataset = TensorDataset(X, y)
loader = DataLoader(dataset, batch_size=32, num_workers=1)
loaders = {"train": loader, "valid": loader}

# model, criterion, optimizer, scheduler
model = torch.nn.Linear(num_features, 1)
criterion = torch.nn.MSELoss()
optimizer = torch.optim.Adam(model.parameters())
scheduler = OneCycleLRWithWarmup(optimizer, num_steps = 6,lr_range=(2e-3,1e-3), init_lr = 1e-3, warmup_fraction = 0.5)

# model training
runner = SupervisedRunner()
runner.train(
    model=model,
    criterion=criterion,
    optimizer=optimizer,
    scheduler=scheduler,
    loaders=loaders,
    logdir="./logdir",
    num_epochs=8,
    verbose=True,
)
```

Then run tensorboard 

```
load_ext tensorboard 
tensorboard --logdir 'logdir/
```

Below is the lr plot I get using tensorboard: 

<img width="992" alt="Screenshot 2020-06-23 at 2 53 31 AM" src="https://user-images.githubusercontent.com/16884938/85324867-34930f80-b4fd-11ea-94fb-da62c801df94.png">

**Comment by Scitator** [comment]
Dear @lokeshkvn ,
looks like I found the solution,
You need to change this line
https://github.com/catalyst-team/catalyst/blob/master/catalyst/core/callbacks/scheduler.py#L121
to 
```python
self._scheduler.recalculate(
    loader_len=runner.loader_len, current_step=runner.epoch - 1
)
```

Would you like to make a PR with this fix?

**Comment by lokeshkvn** [comment]
Thanks I will make a PR soon with the above fix. :smiley:
